IPython/core/shellapp.py

```diff
@@ -312,7 +312,7 @@ def _exec_file(self, fname, shell_futures=False):
                 # behavior.
                 with preserve_keys(self.shell.user_ns, '__file__'):
                     self.shell.user_ns['__file__'] = fname
-                    if full_filename.endswith('.ipy'):
```

Comment:
Tip: endswith support multiple values, so you could do `full_filename.endswith(('.ipy', 'ipynb', '.py'))`...

Reply:
👍